docs/GETTING_STARTED.md

```diff
@@ -1,288 +1,247 @@
-**fuse** - confirm FUSE is installed for mounting userspace filesystems with `fusermount -V`
-(optional, enables mounting without root access)
+- `soci`: the CLI tool used to build/manage SOCI indices.
+- `soci-snapshotter-grpc`: the daemon (a containerd snapshotter plugin) used for lazy loading.
 
-## Start a local ORAS registry
-Most OCI registries do not currently support the "referrers" feature.
-SOCI relies on this feature to link its indices and manifests.  The [ORAS
-project](https://oras.land/) tries to fill this gap; in their words:
+Many `soci` CLI commands need to be run as `sudo`, because the metadata is saved
+in directories that a non-root user often does not have access to.
 
-> Registries are evolving as generic artifact stores. To enable this goal, the ORAS project provides a way to push and pull OCI Artifacts to and from OCI Registries.
+Currently to get the binaries, we need to build the project from source after cloing the repo:
 
-[The longer term solution for this issue is using the OCI Reference
-Types](https://github.com/awslabs/soci-snapshotter/issues/42).  For now we are
-relying on the ORAS project for the referrers feature.
+```shell
+git clone https://github.com/awslabs/soci-snapshotter.git
+cd soci-snapshotter
+make
+```
 
-Given this we want to start our registry using an OCI image from the ORAS 
-project.  We will run it on port 5000 as an HTTP (**NOT** HTTPS) server without
-certificates.  To do this we will use Docker:
+This builds the project binaries into the `./out` directory. You can install them
+to a `PATH` directory (`/usr/local/bin`) with:
 
-```
-docker run -d -p 5000:5000 --restart=always --name registry ghcr.io/oci-playground/registry:v3.0.0-alpha.1
+```shell
+sudo make install
 ```
 
-Since this is just a locally run HTTP server we can interact with
-it using standard `curl` commands. To confirm this is running lets
-use the following commands:
+> This doc assumes SOCI binaries are installed into `PATH`. If not, please use
+> the full path of the binaries (e.g. `./out/soci`).
 
-```
-docker ps
-```
+## Push an image to your registry
 
-which should yield a running container at 0.0.0.0:5000 for the ORAS image. Now
-to check that we can interact with the server run the following:
-```
-curl http://localhost:5000/v2/_catalog
-```
-This should return with `{"repositories":[]}` given that we haven't put anything
-in there yet
+In this document we will use `rabbitmq` from DockerHub `docker.io/library/rabbitmq:latest`.
+We use [AWS ECR](https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-console.html)
+as the public registry for demonstration. Other OCI 1.0 compatible registries should
+also work.
 
-## Push an image to the registry using ctr
-First things first, let's choose our favorite image.  For the purpose of this
-document we will use the latest image of rabbitmq from DockerHub, aptly named
-`docker.io/library/rabbitmq:latest`.
+> You need to change `REGISTR` to your own registry.
```

Comment:
typo: `REGISTR` -> `REGISTRY`

Also, this statement seems misleading. If we're assuming our customers are using ECR, shouldn't they just need to change either the `AWS_ACCOUNT` and `AWS_REGION`? 

I think this section is also missing that we're expecting:
a) That you have AWS credentials that can access ECR
b) That you have already created a repo in ECR.

Reply:
> Also, this statement seems misleading. If we're assuming our customers are using ECR, shouldn't they just need to change either the AWS_ACCOUNT and AWS_REGION

If a user uses ECR, he still needs to run the following commands `export ECR_PASS=...` and `export REGISTRY=`. `AWS_ACCOUNT` and `AWS_REGION` are just helper env variables to construct `REGISTRY`. I don't think `export REGISTRY=<AWS_ACCOUNT>.dkr.ecr.<AWS_REGION>.amazonaws.com` is more clear and is actually less convenient (i.e. user has to manually remove `<AWS_ACCOUNT>` and add his own account.

It's unavoidable to have some registry-specific code. For example, if using dockerhub, this snippet will become something like:

```shell
export DOCKERHUB_ACCOUNT=foobar
export DOCKERHUB_PASS=xxx
export REGISTRY=docker.io/$DOCKERHUB_ACCOUNT
...
sudo ctr i push --user $DOCKERHUB_ACCOUNT:DOCKERHUB_PASS --platform linux/amd64 $REGISTRY/rabbitmq:latest
```

any suggestion on how to make this less misleading?

> a) That you have AWS credentials that can access ECR
> b) That you have already created a repo in ECR.

Added in note. I think it's sufficient to call out these with links other than adding those steps in code. There're multiple different ways of doing this.

Reply:
If we're showing the ECR flow, we should focus on the ECR flow. e.g.

```
This assumes you have created an ECR repo called rabbitmq and have credentials available to the AWS CLI. You just need to update `AWS_ACCOUNT` and `AWS_REGION`. If you are using a different registry, you will need to set `REGISTRY` and `REGISTRY_PASSWORD` appropriately.
```

Also, I think we should rename `ECR_PASS` to `REGISTRY_PASSWORD`.

Reply:
updated